RecoLocalTracker/Phase2TrackerRecHits/plugins/alpaka/Phase2OTRecHitsSoAConverter.cc

```diff
@@ -0,0 +1,276 @@
+            stripHitsHost.view()[idx].yerrLocal() = recHit.localPositionError().yy();
+            auto globalPosition = det->toGlobal(recHit.localPosition());
+            double gx = globalPosition.x() - bs.x0();
+            double gy = globalPosition.y() - bs.y0();
+            double gz = globalPosition.z() - bs.z0();
+            stripHitsHost.view()[idx].xGlobal() = gx;
+            stripHitsHost.view()[idx].yGlobal() = gy;
+            stripHitsHost.view()[idx].zGlobal() = gz;
+            stripHitsHost.view()[idx].rGlobal() = sqrt(gx * gx + gy * gy);
+            stripHitsHost.view()[idx].iphi() = unsafe_atan2s<7>(gy, gx);
+            stripHitsHost.view()[idx].chargeAndStatus().charge = 0;
+            stripHitsHost.view()[idx].chargeAndStatus().status = {false, false, false, false, 0};
+            stripHitsHost.view()[idx].clusterSizeX() = -1;
+            stripHitsHost.view()[idx].clusterSizeY() = -1;
+            stripHitsHost.view()[idx].detectorIndex() = modulesInPixel_ + offset;
+            LogDebug("Phase2OTRecHitsSoAConverter")
+                << "Local (x, y) with (xx, yy) --> (" << recHit.localPosition().x() << ", "
+                << recHit.localPosition().y() << ") with (" << recHit.localPositionError().xx() << ", "
+                << recHit.localPositionError().yy() << ")" << '\n'
+                << "Global           (x, y, z) --> (" << globalPosition.x() << ", " << globalPosition.y() << ", "
+                << globalPosition.z() << ")" << '\n'
+                << "Corrected Global (x, y, z) --> (" << gx << ", " << gy << ", " << gz << ")" << '\n'
+                << gx << '\n';
+          }
+        }
+      }
+    }
+    stripHitsModuleView[orderedModules_.size()].moduleStart() =
+        cumulativeHitPerModule[orderedModules_.size() - 1] + nPixelHits;
+
+#ifdef HITS_DEBUG
+    int current = 0;
+    for (int h = 0; h < stripHitsHost.view().metadata().size(); ++h) {
+      auto idx = stripHitsHost.view()[h].detectorIndex();
+      std::cout << h << " detectorIndexInSoA: " << idx << std::endl;
+      assert(idx >= current);
+      current = idx;
+    }
+    for (int h = 0; h < stripHitsModuleView.metadata().size(); ++h) {
+      std::cout << h << " -> " << stripHitsModuleView[h].moduleStart() << std::endl;
+    }
+#endif
+
+    auto moduleStartView = cms::alpakatools::make_host_view<uint32_t>(stripHitsModuleView.moduleStart(),
+                                                                      stripHitsModuleView.metadata().size());
+    HMSstorage moduleStartVec(stripHitsModuleView.metadata().size());
+
+    // Put in the event the hit module start vector.
+    // Now, this could  be avoided having the Host Hit SoA
+    // consumed by the downstream module (converters to legacy formats).
+    // But this is the common practice at the moment
+    // also for legacy data formats.
+    alpaka::memcpy(queue, moduleStartVec, moduleStartView);
```

Comment:
If I parsed the code correctly, both source and destination are in host memory, and you could use a regular `std::memcpy`.